admin/ops/setup_dev_bucket.py

```diff
@@ -0,0 +1,342 @@
+    ds: str,
+):
+    """
+    Within a dev bucket, resolves/copies the dataset currently in "latest"
+    into its versioned folder under "datasets/{dataset_id}"
+    """
+    os.environ["RECIPES_BUCKET"] = target_bucket
+    importlib.reload(configuration)
+    assert configuration.RECIPES_BUCKET != PROD_RECIPES_BUCKET
+    input = recipes.Dataset(name=ds, version="latest")
+    resolved = recipes.Dataset(name=ds, version=recipes.get_latest_version(ds))
+    os.environ["RECIPES_BUCKET"] = PROD_RECIPES_BUCKET
+    importlib.reload(configuration)
+    assert target_bucket.startswith(DEV_BUCKET_PREFIX)
+    copy_folder_to_dev_bucket(
+        source_bucket=target_bucket,
+        target_bucket=target_bucket,
+        source_path=input.s3_folder_key("datasets") + "/",
+        target_path=resolved.s3_folder_key("datasets") + "/",
+    )
+
+
+def resolve_latest_publish(target_bucket: str, data_product: str):
+    """
+    Within a dev bucket, resolves/copies the data product currently in "publish/latest"
+    into its versioned folder within "{product}/publish"
+    """
+    os.environ["PUBLISHING_BUCKET"] = target_bucket
+    importlib.reload(configuration)
+    assert configuration.PUBLISHING_BUCKET != PROD_PUBLISHING_BUCKET
+    input = PublishKey(product=data_product, version="latest")
+    latest = publishing.get_latest_version(data_product)
+    assert latest
+    resolved = PublishKey(product=data_product, version=latest)
+    os.environ["PUBLISHING_BUCKET"] = PROD_PUBLISHING_BUCKET
+    importlib.reload(configuration)
+    copy_folder_to_dev_bucket(
+        source_bucket=target_bucket,
+        target_bucket=target_bucket,
+        source_path=input.path + "/",
+        target_path=resolved.path + "/",
+    )
+
+
+def clone_recipe(
+    target_bucket: str,
+    dataset_id: str,
+    version: str = "latest",
+):
+    """
+    Given a dataset id, copy it from edm-recipes
+    If provided version is "latest", also resolve it to its versioned folder
+    """
+    key = DatasetKey(name=dataset_id, version=version)
+    copy_folder_to_dev_bucket(
+        source_bucket=PROD_RECIPES_BUCKET,
+        target_bucket=target_bucket,
+        source_path=f"{key.s3_path("datasets")}/",
+        target_path=f"{key.s3_path("datasets")}/",
+    )
+    if version == "latest":
+        resolve_latest_recipe(target_bucket, dataset_id)
+
+
+def clone_recipes_latest(
+    target_bucket: str,
+    include: list[str] | None = None,
+    exclude: list[str] | None = None,
+):
+    """
+    Clone latest versions of various recipe datasets from production bucket to dev bucket
+    By default, targets all datasets
+    Specific datasets can be excluded or included with `include` or 'exclude_dataset`
+    If `include` is specified, only the specified datasets are cloned
+    """
+    if include and exclude:
+        raise ValueError("Only one of `include` and `exclude` may be provided")
+    datasets = get_subfolders(bucket=PROD_RECIPES_BUCKET, prefix="datasets/")
```

Comment:
should we use `recipes.get_all_versions`, to encapsulate folder structure (`datasets/` specifically) within the recipes module?

Reply:
That would be something like `get_all_datasets` which currently doesn't exist, but I could make that function in `recipes`

Reply:
probably ideal, but also super minor